core/ts/Poll.ts

```diff
@@ -295,20 +291,24 @@ export class Poll implements IPoll {
    * @param message - The message to top up the voice credit balance
    */
   public topupMessage = (message: Message): void => {
-    assert(message.msgType == BigInt(2));
+    assert(message.msgType == BigInt(2), "A Topup message must have msgType 2");
+
     for (const d of message.data) {
-      assert((d as bigint) < SNARK_FIELD_SIZE);
+      assert(d < SNARK_FIELD_SIZE, "The message data is not in the correct range");
     }
-    this.messages.push(message);
     const padKey = new PubKey([
```

Comment:
I'm curious about why we  need `padKey` here and how are they chosen?

Reply:
https://github.com/privacy-scaling-explorations/maci/blob/dev/contracts/contracts/crypto/SnarkConstants.sol#L12-L23

here is explained how it's chosen. I'll have to dig a bit deeper as to why it's needed, my immediate guess is that we use it as we need to pass in a public key when publishing a topup message (which is not encrypted), and same for when we push a zero leaf in the message tree on chain